### PR TITLE
Grouse, Pheasants, Turkeys, and Stripped Wood

### PR DESCRIPTION
**Pull request by Therighthon**
Adds Grouse, Pheasants, Turkeys, and their respective meat types
Adds custom side textures to stripped wood blocks

### CODE REVIEW

File: src/main/java/net/dries007/tfc/client/ClientEventHandler.java [hunk 1]
```diff
@@ -19,11 +19,7 @@
 import net.minecraft.client.gui.screens.MenuScreens;
 import net.minecraft.client.gui.screens.inventory.ContainerScreen;
 import net.minecraft.client.gui.screens.inventory.CraftingScreen;
-import net.minecraft.client.model.BoatModel;
-import net.minecraft.client.model.GoatModel;
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/client/ClientEventHandler.java]: please reimport the tfc code style, we are not using wildcard imports anymore so this should be reversed

File: src/main/java/net/dries007/tfc/client/render/entity/SimpleMobRenderer.java [hunk 2]
```diff
@@ -95,6 +106,12 @@ public Builder(EntityRendererProvider.Context ctx, Function<ModelPart, M> model,
             this.name = name;
         }
 
+        public SimpleMobRenderer.Builder<T, M> hasMaleTexture()
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/client/render/entity/SimpleMobRenderer.java]: why?

File: src/main/java/net/dries007/tfc/client/render/entity/SimpleMobRenderer.java [hunk 3]
```diff
@@ -27,17 +30,24 @@
...
+    }
+    public SimpleMobRenderer(EntityRendererProvider.Context ctx, M model, String name, float shadow, boolean flop, float scale, boolean hasBabyTexture, boolean hasMaleTexture, boolean itemInMouth, @Nullable Function<T, ResourceLocation> textureGetter)
     {
         super(ctx, model, shadow);
         doesFlop = flop;
         texture = Helpers.animalTexture(name);
         babyTexture = hasBabyTexture ? Helpers.animalTexture(name + "_young") : null;
-        this.textureGetter = textureGetter != null ? textureGetter : e -> babyTexture != null && e.isBaby() ? babyTexture : texture;
+        maleTexture = hasMaleTexture ? Helpers.animalTexture(name + "_male") : null;
+        this.textureGetter = textureGetter != null ? textureGetter : e -> babyTexture != null && e.isBaby() ? babyTexture : maleTexture != null && (e instanceof WildAnimal) && ((WildAnimal) e).isMale() ? maleTexture : texture;
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/client/render/entity/SimpleMobRenderer.java]: this is getting too complicated. you can just use the texture getter. also check GenderedRenderAnimal and not WidlAnimal if that's what you're doing.

File: resources/data.py [hunk 4]
```diff
@@ -1000,6 +1004,9 @@ def needs_tool(_tool: str) -> str:
     rm.data(('tfc', 'fauna', 'ocelot'), fauna(climate=climate_config(min_rain=300, max_rain=500, min_temp=15, max_temp=30, min_forest='edge')))
     rm.data(('tfc', 'fauna', 'deer'), fauna(climate=climate_config(min_rain=130, max_rain=400, min_temp=-15, max_temp=25, min_forest='edge')))
     rm.data(('tfc', 'fauna', 'moose'), fauna(climate=climate_config(min_rain=150, max_rain=300, min_temp=-15, max_temp=10, min_forest='edge')))
+    rm.data(('tfc', 'fauna', 'grouse'), fauna(climate=climate_config(min_rain=150, max_rain=400, min_temp=-12, max_temp=13)))
```
Comment (eerussianguy) [inline resources/data.py:1009]: need book entries for these guys

**Review by eerussianguy — CHANGES_REQUESTED** [review]

**Comment by eerussianguy** [comment]
also please merge master so that resource validation works on this pr, there was an issue earlier

**Comment by eerussianguy** [comment]
also need to rerun texture generation

File: src/main/java/net/dries007/tfc/common/entities/GenderedRenderAnimal.java [hunk 5]
```diff
@@ -6,9 +6,19 @@
...
+
+import net.dries007.tfc.util.Helpers;
+
 public interface GenderedRenderAnimal
 {
     boolean displayMaleCharacteristics();
 
     boolean displayFemaleCharacteristics();
+
+    public static <T> ResourceLocation getGenderedTexture(GenderedRenderAnimal animal, String name)
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/common/entities/GenderedRenderAnimal.java]: this is duplicated in WildAnimal? also this should probably be next to the `animalTexture` method in `Helpers`.

File: src/main/java/net/dries007/tfc/common/entities/TFCEntities.java [hunk 6]
```diff
@@ -42,11 +42,7 @@
 import net.dries007.tfc.common.entities.livestock.pet.TFCCat;
 import net.dries007.tfc.common.entities.predator.FelinePredator;
 import net.dries007.tfc.common.entities.predator.Predator;
-import net.dries007.tfc.common.entities.prey.Pest;
-import net.dries007.tfc.common.entities.prey.Prey;
-import net.dries007.tfc.common.entities.prey.TFCFox;
-import net.dries007.tfc.common.entities.prey.TFCPanda;
-import net.dries007.tfc.common.entities.prey.TFCRabbit;
+import net.dries007.tfc.common.entities.prey.*;
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/common/entities/TFCEntities.java]: missed this

File: src/main/java/net/dries007/tfc/common/entities/GenderedRenderAnimal.java (hunk 5 above)
Comment (eerussianguy) [reply src/main/java/net/dries007/tfc/common/entities/GenderedRenderAnimal.java]: for esoteric performance reasons, can this be made into

```
final ResourceLocation male = Helpers.animalTexture(name + "_male")
final ResourceLocation female = Helpers.animalTexture(name + "_female")
return animal -> animal.displayMaleCharacteristics() ? male : female;
```

**Review by eerussianguy — CHANGES_REQUESTED** [review]

File: src/main/java/net/dries007/tfc/common/entities/GenderedRenderAnimal.java (hunk 5 above)
Comment (Therighthon) [reply src/main/java/net/dries007/tfc/common/entities/GenderedRenderAnimal.java]: Changed to

`final ResourceLocation male = Helpers.animalTexture(name + "_male");`
`final ResourceLocation female = Helpers.animalTexture(name + "_female");`
`return animal.displayMaleCharacteristics() ? male : female;`

**Review by eerussianguy — APPROVED** [review]

**Pull request merged**